pkg/tcpip/network/ipv4/ipv4_frag_test.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2018 The gVisor Authors.                                      
+//                                                                         
+// Licensed under the Apache License, Version 2.0 (the "License");         
+// you may not use this file except in compliance with the License.        
+// You may obtain a copy of the License at                                 
+//                                                                         
+//     http://www.apache.org/licenses/LICENSE-2.0                          
+//                                                                         
+// Unless required by applicable law or agreed to in writing, software     
+// distributed under the License is distributed on an "AS IS" BASIS,       
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and     
+// limitations under the License.
+
+package ipv4_test
+
+import (
+	"testing"
+	"time"
+    "fmt"
+
+	"gvisor.dev/gvisor/pkg/tcpip/buffer"
+	"gvisor.dev/gvisor/pkg/tcpip/header"
+	"gvisor.dev/gvisor/pkg/tcpip/network/fragmentation"
+)
+
+const (
+	localIpv4Addr  = "\x0a\x00\x00\x01"
+	remoteIpv4Addr = "\x0a\x00\x00\x02"
+)
+
+// vv is a helper to build VectorisedView from different strings.
+func vv(size int, pieces ...string) buffer.VectorisedView {
+	views := make([]buffer.View, len(pieces))
+	for i, p := range pieces {
+		views[i] = []byte(p)
+	}
+
+	return buffer.NewVectorisedView(size, views)
+}
+
+func TestReassemblingTimeout(t *testing.T) {
+    fmt.Println("\n ----------------STARTED REASSEMBLY TIMEOUT TEST----------------- \n")
+	totalLen := header.IPv4MinimumSize + 30
+	view := buffer.NewView(totalLen)
+	ip := header.IPv4(view)
+	ip.Encode(&header.IPv4Fields{
+		IHL:         header.IPv4MinimumSize,
+		TotalLength: uint16(totalLen),
+		TTL:         20,
+		Protocol:    10,
+		SrcAddr:     remoteIpv4Addr,
+		DstAddr:     localIpv4Addr,
+	})
+	v1 := view.ToVectorisedView()
+	timeout := time.Second
+	f := fragmentation.NewFragmentation(1024, 512, timeout)
+	// Send first fragment with id = 0, first = 0, last = 0, and more = true.
+	f.Process(0, 0, 0, true, vv(1, "0"), v1.First())
+	// Sleep more than the timeout.
+	time.Sleep(35 * timeout)
```

Comment:
why do we want to sleep in 35 times more than timeout?

Reply:
taken care